### ISSUE TITLE
Allow more control of TextInputControls in EditableText

### ISSUE THREAD

**Issue by maeddin** [flutter/flutter#157859]
### Use case

Some apps need the option of using their own keyboard for security reasons. You often do not want other software or hardware keyboards to be able to track what is currently being entered in a text field.
Possible use cases are, for example, banking apps or other apps that have to meet certain security requirements.

We are currently unable to publish our Flutter app because a requirement for this is that external input is not allowed. The only possibility at the moment is to implement `EditableText` and `TextField`/`TextFormField` ourselves.

### Proposal

It could be made possible for `EditableTexts` or `TextFields` to control which `TextInputs` can modify them or receive updates from `TextEditingValues` or similar. For example, you could give each `TextInputConnection` the option of defining its own filter function for all permitted `TextInputControls`. Alternatively you could at least provide the option to prohibit external `TextInputs` such as `_PlatformTextInputControl.instance`.

**Comment by LongCatIsLooong** [comment]
To use in-app keyboards, you can disable software keyboard using [`TextInputType.none`](https://main-api.flutter.dev/flutter/services/TextInputType/none-constant.html). Does that work for you?

**Comment by maeddin** [comment]
> To use in-app keyboards, you can disable software keyboard using [`TextInputType.none`](https://main-api.flutter.dev/flutter/services/TextInputType/none-constant.html). Does that work for you?

It's not primarily about using your own keyboard. You can do that relatively easily. Nevertheless, in this case the keyboard of the operating system is informed of every change, which is a security vulnerability that is sometimes not allowed. 

In the following two places, each connected keyboard is informed of a change regardless of the `TextInputType`. It would therefore be useful if you could somehow control which `TextInputControls` are allowed to access a `TextField`.

https://github.com/flutter/flutter/blob/86fc1fd47951eab6031c1c7e602d8499bf33216e/packages/flutter/lib/src/services/text_input.dart#L1991-L1995

https://github.com/flutter/flutter/blob/86fc1fd47951eab6031c1c7e602d8499bf33216e/packages/flutter/lib/src/services/text_input.dart#L2057-L2068

**Comment by LongCatIsLooong** [comment]
It looks like the platform text input control is considered "special" atm: 

https://github.com/flutter/flutter/blob/86fc1fd47951eab6031c1c7e602d8499bf33216e/packages/flutter/lib/src/services/text_input.dart#L1649-L1659

It seems to make sense to be able to remove them as well. But since you want to prevent communication between the text field & the on-screen / physical keyboard, does it work if you set the text field to readOnly (which prevents the text field from establish any text input connections) and use the TextEditingController to control the input? I haven't tried that and things like blinking cursors may not work.

**Comment by maeddin** [comment]
> I haven't tried that and things like blinking cursors may not work.

@LongCatIsLooong Yes, this does not work for the reason you mentioned: The cursor is not displayed correctly. Even if it would work, it would still be useful if you could implement this use case without self-implementation that bypasses `TextInput` with a `TextEditingController` .

**Comment by maeddin** [comment]
There are also other reasons why custom virtual keyboards do not work with `EditableText`. For example, scrolling to the current `EditableText` only works with the native keyboard, as scrolling is based on `View.of(context)` here.

https://github.com/flutter/flutter/blob/86fc1fd47951eab6031c1c7e602d8499bf33216e/packages/flutter/lib/src/widgets/editable_text.dart#L4133-L4150

So it's generally difficult to use your own keyboards in Flutter at the moment.
